packages/graphql-mini-transforms/src/document.ts

```diff
@@ -81,6 +104,22 @@ export function toSimpleDocument<
   };
 }
 
+export function formatDocument(document: DocumentNode, format: OutputFormat) {
+  switch (format) {
+    case 'document': {
+      return document;
+    }
+    case 'simple': {
+      return toSimpleDocument(document);
+    }
+    case 'simple-persisted': {
+      const simple = toSimpleDocument(document);
+      delete (simple as any).source;
```

Comment:
Creating the source key within `toSimpleDocument` then deleting it here feels a little suprising.

What if `toSimpleDocument`'s signature was changed to `toSimpleDocument(document, omitSource = false), and then it's up to toSimpleDocument to add/omit the source.


It looks like SimpleDocument has `source` as a required parameter. Will things get weird if the type says that a `source` key will be present, but at runtime the `source` key is absent? Would setting the source to be present but an empty string sidestep violating the type?